packages/semcom-demo/src/app/connect/connect.guard.ts

```diff
@@ -0,0 +1,49 @@
+import { CanActivate, Resolve, Router, UrlTree } from '@angular/router';
+import { ISessionInfo, handleIncomingRedirect } from '@inrupt/solid-client-authn-browser';
+import { map, take } from 'rxjs/operators';
+import { AppState } from '../app.reducers';
+import { Injectable } from '@angular/core';
+import { Location } from '@angular/common';
+import { Observable } from 'rxjs';
+import { ProviderConnected } from './connect.actions';
+import { Store } from '@ngrx/store';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConnectGuard implements CanActivate, Resolve<void> {
+
+  sessionInfo$: Observable<ISessionInfo|null> = this.store.select(state => state.connectState.sessionInfo);
+
+  constructor(private store: Store<AppState>, private router: Router, private location: Location) {}
+
+  resolve(): void {
```

Comment:
Very nice; great idea to implement this in a resolve function. 

Just a little suggestion: wouldn't it be clearer to define split this off in a separate class, given that a `Resolver` is not conceptually the same as a `Guard`?

Reply:
Actually, Angular docs describe it as a guard. That's wy I thought it nice to put all connect-related guard-logic in the same class.